src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2091,7 +2092,7 @@ because xbuild doesn't support framework reference assemblies.
 </Target>
 	
 <Target Name="_LinkAssemblies"
-  DependsOnTargets="_ResolveAssemblies;_CreatePackageWorkspace;_GenerateJniMarshalMethods;_LinkAssembliesNoShrink;_LinkAssembliesShrink" />
+  DependsOnTargets="_ResolveAssemblies;_CreatePackageWorkspace;_GenerateJniMarshalMethods;$(_BeforeLinkAssembliesShrink);_LinkAssembliesNoShrink;_LinkAssembliesShrink" />
```

Comment:
I think you can remove this line and just use `$(_BeforeLinkAssemblies)`. If the target needs to know if it's "no shrink", then it can just check `$(AndroidLinkMode)`.